### PR TITLE
Exposing message room to jobs

### PR DESCRIPTION
**Pull request by tachyon2901**
exposing bot and message room to jobs so that they can send messages to other rooms and use other hubot functionalities

- [x] Semicolons - and other formatting
- [x] Test every function 
- [x] Const every variable that won’t change.
- [x] Abstract out code that is being used multiple times as a separate function
- [x] Don't use moment, request etc 
- [x] Don’t async functions that don’t need to be async
- [x] Add conditional chaining (?) when fields could not exist in objects
- [x] Maintain consistent casing for all variables, functions, files etc.
- [x] Remove extra spaces when they have been added
- [x] Always normalize casing for data when matching
- [x] If we can clues.reject use that instead of throw to prevent a large stack trace

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -64,6 +70,9 @@ module.exports = function(bot){
           {
             send: function(msg){
               sendMessage(bot,message,msg);
+            },
+            messageRoom: function (msg, room) {
```
Comment (darogina) [inline index.js]: Instead of creating a brand new function why not just modify the existing send function to take in an optional room and then modify the existing `sendMessage` function to also take in an optional room.  the sendMessage function is calling off to bot.messageRoom() already so lets just piggy back off of that.

File: index.js [hunk 2]
```diff
@@ -64,6 +70,9 @@ module.exports = function(bot){
...
+              sendMessagetoRoom(bot, msg, room)
```
Comment (darogina) [inline index.js]: It would also potentially be beneficial to return out the original bot object as well as the send function here.  That will give us flexibility in the future for jobs to interact with other hubot specific functions like pulling from the brain.

File: index.js [hunk 3]
```diff
@@ -18,67 +18,69 @@ module.exports = function(bot){
 
   let scheduler = () => bot.Scheduler || Scheduler;
 
-  var room = process.env.HUBOT_JOB_CHANNEL;
+  var mainRoom = process.env.HUBOT_JOB_CHANNEL;
```
Comment (darogina) [inline index.js]: No reason to be using var.  Can just be const since you won't ever be changing it.

File: index.js [hunk 4]
```diff
@@ -18,67 +18,69 @@ module.exports = function(bot){
...
 
-  function sendMessage(bot, message, msg) {
+  function sendMessage(bot, message, msg, room = mainRoom) {
     bot.messageRoom(room, msg);
-    if (message && message.envelope && message.envelope.room !== room){
+    if (message && message.envelope && message.envelope.room !== mainRoom) {
```
Comment (darogina) [inline index.js]: Why is this changed to mainRoom?  Don't you still want to be checking just room?

Comment (tachyon2901) [reply index.js]: oh yeah - I changed the name of the default `room` variable from `process.env.HUBOT_JOB_CHANNEL;` to `mainRoom` to avoid confusion since a couple of the parameters are now called room. Maybe it makes sense to change the room parameter to `channel` and maintain the variable`room`.

**Review by jpcolomer — APPROVED** [review]
